src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -2470,6 +2470,24 @@ static bool contextElementPreconditionsCheck
   /* Getting the entity in the request (helpful in other places) */
   EntityId* enP = &ceP->entityId;
 
+  /* Checking there aren't duplicate attributes */
```

Comment:
At the end, the checking is done at contextElementPreconditionsCheck() before starting processing the CE. Another alternative would be to keep record of the attributes while they are inserted in the toSet and toPush structures at processAttributeVector() but it is much complex and without a significant gain.

The getId() search makes this check expensive, but there isn't any good way of fighting agains that... except remove the ID functionality istelf, of course ;)

NTC (just an explanation of the fix)